### ISSUE TITLE
[BUG] paths of test files in "playwright test --list" on Windows is screwed up

### ISSUE THREAD

**Issue by iva2k** [microsoft/playwright#15202]
**Context:**
- Playwright Version: 1.23.0 (latest)
- Browser: All
- Extra:

## System:
 - OS: Windows 10 10.0.19044
 - Memory: 956.04 MB / 15.74 GB
## Binaries:
 - Node: 18.1.0 - C:\nodejs\node.EXE
 - Yarn: 1.22.15 - C:\nodejs\yarn.CMD
 - npm: 8.10.0 - C:\nodejs\npm.CMD
## Languages:
 - Bash: 4.4.23 - C:\Program Files\Git\usr\bin\bash.EXE (Windows CMD has same issue, so it is not git/bash)
 
**Describe the bug**

Paths of test files are screwed up in `playwright test --list`.

**Code Snippet**

```bash
playwright test --list
# Listing tests:
#  [chromium] › ..\..\..\..\C:\dev\playwright\playwright-ts1\tests\example.spec.ts:14:3 › New Todo should allow me to add todo items
#  [chromium] › ..\..\..\..\C:\dev\playwright\playwright-ts1\tests\example.spec.ts:37:3 › New Todo should clear text input field when an item is added
# ...
```
Expected normal path without `..\..\..\..\C:\` part.

Additional info:

- Was working correctly in playwright 1.22.2
- playwright-vscode can't run individual test files or open the test with errors, caused by path like "C:\c:\playwright\try\tests\example.spec.ts"

**Comment by rwoll** [comment]
Thanks for the report—this looks like a regression, and will be triaged accordingly.

**Comment by rwoll** [comment]
@iva2k **Can you link me to a repo I can clone and run to see the behavior?** I'll be able to debug more quickly. I tried repro'ing via:

```
> npm init playwright@latest
> npx playwright test --list
```

but the output is expected and the tests run.

Additionally, please elaborate on your file structure:

- Where is the (full) path to your package.json?
- Where is the (full) path to your test files?
- Where is the(full) path where you are running `npx playwright test --list` from?
- What does your playwright config look like if any?
- What is the (full) path to the config, if any?

Thanks!

**Comment by rwoll** [comment]
Ah—just saw the repro you filed in https://github.com/microsoft/playwright/issues/15200. I'm not yet able to repro, but will keep trying. I've closed that issue to keep things centralized here.

**Comment by rwoll** [comment]
I'm able to repro via:

```
> npm init playwright@latest repro
> npx playwright test --list # this works as expected
> # change module field in package.json to `"type": "module"`, and re-run the above test command,
> npx playwright test --list # now we see the odd-looking paths
```

**Comment by rwoll** [comment]
Here's the repro in our test suite: https://github.com/microsoft/playwright/runs/7102600449?check_suite_focus=true#step:8:46 The related PR has a comment with the change that caused the regression.

NB: I can still run the test even though the paths are wrong.

@iva2k If you try the exact commands in https://github.com/microsoft/playwright/issues/15202#issuecomment-1169349401, do they work for you? I noticed in your repro, your `playwright` commands are not prefixed with `npx`. Did you add a global installation of Playwright to your path? Or which Playwright command are you picking up?

**Comment by iva2k** [comment]
@rwoll - I got the same behavior as in https://github.com/microsoft/playwright/issues/15202#issuecomment-1169349401

I have playwright installed globally in pnpm, but not in npm. Using npx in my tests made no difference.

```cmd
C:\dev\playwright\repro>where playwright
C:\dev\playwright\repro\node_modules\.bin\playwright
C:\dev\playwright\repro\node_modules\.bin\playwright.cmd

playwright --version
Version 1.23.0
```

**Comment by rwoll** [comment]
Thanks—we're continue looking at this today. Sorry for the inconvenience! For the time being, does going back to 1.22 unblock you?

**Comment by mxschmitt** [comment]
This should be fixed in 1.23.1 which gets released soon. You can try it now to verify the fix by `npm install @playwright/test@beta`.

**Comment by iva2k** [comment]
Luckily I'm not blocked, just kicking tires on new toolchain. Thanks for support and Playwright is an amazing tool!

**Comment by mxschmitt** [comment]
Closing since it should be fixed. Please file a new issue if you have more bugs to report, thanks!
